### ISSUE TITLE
change the point sizes in plotColData

### ISSUE THREAD

**Issue by elimereu** [alanocallaghan/scater#60]
Hi Davis, 

Thank you for this amazing tool! I always use it in my daily work.
However, I've just installed the new R version and now I ended up with this small issue. 
I'm trying to increase the size of points in a plot called by the function plotColData. 

The exact call is: plotColData(sceset, x = "X" , y = "Y",colour_by="Z", size=2).

In the previous version, with the old function plotPhenoData, I could fix the problem in this way:

plotPhenoData(sceset, aes(x = X , y = Y , colour = Z),size=2) , 

but now it doesn't seem to work. 

Thank you in advance.

Bests, 

Elisabetta

**Comment by LTLA** [comment]
It would have helped if you described exactly what the problem was. Running the command gives me:

```
Error in .choose_vis_values(se, size_by, mode = mode, search = "any",  : 
  invalid value of 'by' supplied
```

... which is due to the fact that `size=` somehow expands to `size_by=`, for reasons unclear to me - I will follow this up elsewhere. This can be relatively easily fixed by forcing `size_by=NULL`:

```r
example(plotColData)
plotColData(example_sce, y = "total_features_by_counts", 
    x = "log10_total_counts", size_by=NULL, size=5)
```

... which will force the argument passing mechanism to behave.

**Comment by LTLA** [comment]
Having thought about it, I guess the argument matching behaviour is not all that surprising. The point size argument `size` is passed via `...` in `plotColData`, but `size_by=` is a top-level argument in `plotColData`'s function definition. This means that when R does argument matching, it only sees `size_by` (as the `size=` option is deferred until the remaining arguments are passed to `...`). As such, R will expand `size` to `size_by` in the absence of any other options, leading to the odd behaviour observed above.

The best solution is probably to avoid having shorter arguments that can't be disambiguated from longer ones. I'll change `size` to `point_size` (and `alpha` to `point_alpha`) in the latest version.

**Comment by elimereu** [comment]
Hi Aaron, 

thank you a lot for these suggestions.
I got the problem!

I really liked the previous behaviour of plotPhenoData because I could “interact” in some way with ggplot2. 
More specifically, for example I could use the aes function inside, which was allowing me to set colour and size
without define an external data.frame. 
For example: 

plotPhenoData(sceset, aes(x = X , y = Y , colour = Z),size=2,alpha=0.8)

now is: 

df=as.data.frame(colData(secret))

ggplot(df,aes(x=X,y=Y))+geom_point(aes(colour = Z),size=2,alpha=0.8). 


It’s true I could do something like: 

plotColData(sceset, x = “X”, y = “Y", colour_by = “Z")


but if I want to change default colours or style is more tricky or I’m not sure if it directly possible.. 

Thank you again!

Elisabetta

**Comment by LTLA** [comment]
I'm not sure I understand the problem; your second command should do what you want. If you want to change the colours or style, you should be able to do so by adding further commands to the _ggplot_ object (e.g., `+ scale_color_gradientn()` or something similar). If this is not enough, and you need further customization... then it is probably easier to create the _ggplot_ object manually.

The problem with the old `plotPhenoData` was precisely that it exposed the _ggplot_ mechanics to the user. If the user needs to know _ggplot_ in order to use `plotPhenoData`, they might as well use _ggplot_ directly.

**Comment by elimereu** [comment]
Yes, you understood perfectly.. 
The problem is that when I add those arguments, they are not working, the plot doesn’t change.
It’s not a big problem anyway.


Thanks, 

Elisabetta

**Comment by LTLA** [comment]
Your specific problem (with `size`) has been fixed in #61, and will get merged in sometime this week.

**Comment by elimereu** [comment]
Perfect!
Thanks again!

Elisabetta
